frontend/app/components/banner/BannerBuilder.tsx

```diff
@@ -61,8 +59,10 @@ export function BannerBuilder({ onRefresh }: Props) {
 }
 
 function ContentBuilder({ onRefresh }: Props) {
-  const { actions: uiActions, state: uiState } = useUI()
-  const profile = toolState.currentConfig as BannerConfig
+  const uiState = useUIState()
+  const uiActions = useUIActions()
+  const snap = useCurrentConfigSnapshot()
+  const profile = toolState.currentConfig
```

Comment:
Can we add another hook here for this? I would rather we not access `toolState` directly anywhere, and rely on hooks (essentially getters) to maintain abstraction.